bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/query/pipeline/Frame.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for
+ * the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.runtime.internal.query.pipeline;
+
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_QUERY_PKG_ID;
+
+/**
+ * Represents a frame that wraps elements as Ballerina records (BMap).
+ *
+ * @since 2201.13.0
+ */
+public class Frame {
```

Comment:
We can remove this class and use  ValueCreator.createRecordValue(BALLERINA_QUERY_PKG_ID, "_Frame"). This will add adddtional method calls and new objects